File: Chess/src/main/java/game/gui/Table.java
package game.gui;

import game.board.Board;
import game.piece.Team;
import game.Data;
import msg.Message;
import side_client.Client;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class Table {
    public JFrame gameFrame;
    public BoardPanel boardPanel;
    public Board chessBoard;
    public MainMenu mainMenu;
    public InGameBottomMenu bottomGameMenu;
    public Client client;

    public Table() {
        this.gameFrame = new JFrame("Chess");
        this.gameFrame.setLayout(new BorderLayout());
        this.gameFrame.setSize(Data.OUTER_FRAME_DIMENSION);
        this.mainMenu = new MainMenu();
        this.client = new Client(this);
        this.client.connect("127.0.0.1", 4000);
        if (this.client.socket == null) {
            JOptionPane.showMessageDialog(null, "Servera bağlanılamadı");
            System.exit(0);
        }
        createMainMenu();
        this.gameFrame.setVisible(true);
    }

    public void createMainMenu() {
        this.mainMenu.getInfoLBL().setText("");
        this.mainMenu.getInfoLBL().setVisible(false);
        this.mainMenu.getPlayBTN().addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                if (!client.isPaired) {
                    mainMenu.getInfoLBL().setVisible(true);
                    mainMenu.getInfoLBL().setText("Matching...");
                    mainMenu.getPlayBTN().setEnabled(false);
                    Message msg = new Message(Message.MessageTypes.PAIRING);
                    msg.content = "ESLESME";
                    client.send(msg);
                }
                if (client.isPaired) {
                    mainMenu.getInfoLBL().setText("Matched");
                    mainMenu.getInfoLBL().setText("Game is starting...");
                    mainMenu.getPlayBTN().setEnabled(true);
                    mainMenu.getInfoLBL().setText("");
                    mainMenu.getInfoLBL().setVisible(false);
                    createGamePanel();
                }
            }
        });
        this.gameFrame.add(mainMenu, BorderLayout.CENTER);
    }

    public void createGamePanel() {
        this.gameFrame.remove(mainMenu);
        this.chessBoard = new Board();
        this.boardPanel = new BoardPanel(this.chessBoard, this.client);
        this.bottomGameMenu = new InGameBottomMenu();
        this.bottomGameMenu.getPlayersColorLBL().setText("Your color is " + this.client.team.toString());
        if (this.client.team == Team.WHITE) {
            this.bottomGameMenu.getTurnLBL().setText("Your Turn");
            this.bottomGameMenu.getTurnLBL().setForeground(Color.GREEN);
        } else {
            this.bottomGameMenu.getTurnLBL().setText("Enemy Turn");
            this.bottomGameMenu.getTurnLBL().setForeground(Color.RED);
        }
        this.gameFrame.add(boardPanel);
        this.gameFrame.add(this.boardPanel, BorderLayout.CENTER);
        this.gameFrame.add(this.bottomGameMenu, BorderLayout.PAGE_END);

        this.gameFrame.setVisible(true);
    }

    public MainMenu getMainMenu() {
        return mainMenu;
    }

    public InGameBottomMenu getBottomGameMenu() {
        return bottomGameMenu;
    }

    public JFrame getGameFrame() {
        return gameFrame;
    }

    public BoardPanel getBoardPanel() {
        return boardPanel;
    }

    public Board getChessBoard() {
        return chessBoard;
    }
}

File: Chess/src/main/kotlin/hummel/game/gui/Table.kt
package main.kotlin.hummel.game.gui

import main.kotlin.hummel.game.Data
import main.kotlin.hummel.game.board.Board
import main.kotlin.hummel.game.piece.Team
import main.kotlin.hummel.msg.Message
import main.kotlin.hummel.side_client.Client
import java.awt.BorderLayout
import java.awt.Color
import javax.swing.JFrame
import javax.swing.JOptionPane
import kotlin.system.exitProcess

class Table {
	var gameFrame: JFrame = JFrame("Chess")
	var mainMenu: MainMenu
	var client: Client
	lateinit var boardPanel: BoardPanel
	lateinit var chessBoard: Board
	lateinit var bottomGameMenu: InGameBottomMenu

	init {
		gameFrame.layout = BorderLayout()
		gameFrame.size = Data.OUTER_FRAME_DIMENSION
		mainMenu = MainMenu()
		client = Client(this)
		client.connect("127.0.0.1", 4000)
		if (client.socket == null) {
			JOptionPane.showMessageDialog(null, "Servera bağlanılamadı")
			exitProcess(0)
		}
		createMainMenu()
		gameFrame.isVisible = true
	}

	fun createMainMenu() {
		mainMenu.infoLBL!!.text = ""
		mainMenu.infoLBL!!.isVisible = false
		mainMenu.playBTN!!.addActionListener {
			if (!client.isPaired) {
				mainMenu.infoLBL!!.isVisible = true
				mainMenu.infoLBL!!.text = "Matching..."
				mainMenu.playBTN!!.isEnabled = false
				val msg = Message(Message.MessageTypes.PAIRING)
				msg.content = "ESLESME"
				client.send(msg)
			}
			if (client.isPaired) {
				mainMenu.infoLBL!!.text = "Matched"
				mainMenu.infoLBL!!.text = "Game is starting..."
				mainMenu.playBTN!!.isEnabled = true
				mainMenu.infoLBL!!.text = ""
				mainMenu.infoLBL!!.isVisible = false
				createGamePanel()
			}
		}
		gameFrame.add(mainMenu, BorderLayout.CENTER)
	}

	private fun createGamePanel() {
		gameFrame.remove(mainMenu)
		chessBoard = Board()
		boardPanel = BoardPanel(chessBoard, client)
		bottomGameMenu = InGameBottomMenu()
		bottomGameMenu.playersColorLBL.text = "Your color is " + client.team.toString()
		if (client.team === Team.WHITE) {
			bottomGameMenu.turnLBL.text = "Your Turn"
			bottomGameMenu.turnLBL.foreground = Color.GREEN
		} else {
			bottomGameMenu.turnLBL.text = "Enemy Turn"
			bottomGameMenu.turnLBL.foreground = Color.RED
		}
		gameFrame.add(boardPanel)
		gameFrame.add(boardPanel, BorderLayout.CENTER)
		gameFrame.add(bottomGameMenu, BorderLayout.PAGE_END)
		gameFrame.isVisible = true
	}
}